tests/features/custom_page/collection.custom_page.feature

```diff
@@ -0,0 +1,44 @@
+@api
+Feature: "Add custom page" visibility options.
+  In order to manage custom pages
+  As a collection member
+  I need to be able to add "Custom Page" content through UI.
+  Background:
+    Given users:
+    | name            | mail                  | roles         | pass |
+    | Rufus Drumknott | member@example.com    |               | test |
+    And the following collections:
+    | uri                             | name           | description               | logo     |  owner |
+    | http://joinup.eu/collection/foo | Foo Collection | This is a foo collection. | logo.png |        |
+    | http://joinup.eu/collection/bar | Bar Collection | This is a bar connection. | logo.png |        |
+    And the following user memberships:
+    | group_type | group_id                         | member          |
+    | rdf_entity | http://joinup.eu/collection/foo  | Rufus Drumknott |
+
+  Scenario: Check visibility of the "Add custom page" button.
+    When I am logged in as "Rufus Drumknott"
+    And I go to the homepage of the "Foo Collection" collection
+    Then I should see the link "Add custom page"
+    When I click "Leave this collection"
+    And I press the "Confirm" button
+    Then I should not see the link "Add custom page"
+    When I go to the homepage of the "Bar Collection" collection
+    Then I should not see the link "Add custom page"
+    But I should see the button "Join this collection"
+    When I press the "Join this collection" button
+    Then I should see the link "Add custom page"
```

Comment:
The reason of this test is to prove that the button is cached but varies per user 'and' per collection. So the meaning is that when a state changes for one collection (Foo), then if you visit the other collection (Bar), they do not share the same cached block. It is actually a test that the caching occurs correctly.